src/robusta/integrations/slack/sender.py

```diff
@@ -213,16 +219,37 @@ def __to_slack(self, block: BaseBlock, sink_name: str) -> List[SlackBlock]:
             logging.warning(f"cannot convert block of type {type(block)} to slack format block: {block}")
             return []  # no reason to crash the entire report
 
-    def __upload_file_to_slack(self, block: FileBlock, max_log_file_limit_kb: int) -> str:
+    def __upload_file_to_slack(self, block: FileBlock, max_log_file_limit_kb: int) -> Optional[str]:
+        """Upload a file to Slack and return a permalink to it."""
         truncated_content = block.truncate_content(max_file_size_bytes=max_log_file_limit_kb * 1000)
+        filename = block.filename
 
-        """Upload a file to slack and return a link to it"""
-        with tempfile.NamedTemporaryFile() as f:
+        def _upload_temp_file(file_reference) -> Optional[str]:
```

Comment:
Why we need this as inner function? any good reason for that? If not please move it outside of the function.

Reply:
removing duplicate calls of the same 5 lines for each

Reply:
But why it is inner function? why not have it as function on the class?

Reply:
got it